taskvine/src/manager/vine_manager.c

```diff
@@ -780,11 +810,13 @@ static void remove_worker(struct vine_manager *q, struct vine_worker_info *w, vi
 
 	vine_txn_log_write_worker(q, w, 1, reason);
 
-	cleanup_worker(q, w);
-
 	hash_table_remove(q->worker_table, w->hashkey);
 	hash_table_remove(q->workers_with_available_results, w->hashkey);
 
+	recover_worker_temp_files(q, w);
```

Comment:
Make this contingent on an option that can be set with `vine_tune`.
We ware going to compare various options for recovery and you will want to control them from the outside.